cmd/podman/system/df.go

```diff
@@ -143,78 +124,94 @@ func printSummary(reports *entities.SystemDfReport, userFormat string) error {
-	fmt.Print("\nImages space usage:\n\n")
+	fmt.Fprint(w, "Images space usage:\n\n")
 	// convert to dfImage for output
 	dfImages := make([]*dfImage, 0, len(reports.Images))
 	for _, d := range reports.Images {
 		dfImages = append(dfImages, &dfImage{SystemDfImageReport: d})
 	}
-	imageHeaders := "REPOSITORY\tTAG\tIMAGE ID\tCREATED\tSIZE\tSHARED SIZE\tUNIQUE SIZE\tCONTAINERS\n"
+	hdrs := report.Headers(entities.SystemDfImageReport{}, map[string]string{
+		"ImageID":    "IMAGE ID",
+		"SharedSize": "SHARED SIZE",
+		"UniqueSize": "UNIQUE SIZE",
+	})
 	imageRow := "{{.Repository}}\t{{.Tag}}\t{{.ImageID}}\t{{.Created}}\t{{.Size}}\t{{.SharedSize}}\t{{.UniqueSize}}\t{{.Containers}}\n"
-	format := imageHeaders + "{{range . }}" + imageRow + "{{end}}"
-	if err := writeTemplate(w, format, dfImages); err != nil {
+	format := "{{range . }}" + imageRow + "{{end}}"
+	if err := writeTemplate(cmd, w, hdrs, format, dfImages); err != nil {
 		return nil
 	}
 
 	// Containers
-	fmt.Print("\nContainers space usage:\n\n")
+	fmt.Fprint(w, "\nContainers space usage:\n\n")
 
 	// convert to dfContainers for output
 	dfContainers := make([]*dfContainer, 0, len(reports.Containers))
 	for _, d := range reports.Containers {
 		dfContainers = append(dfContainers, &dfContainer{SystemDfContainerReport: d})
 	}
-	containerHeaders := "CONTAINER ID\tIMAGE\tCOMMAND\tLOCAL VOLUMES\tSIZE\tCREATED\tSTATUS\tNAMES\n"
+	hdrs = report.Headers(entities.SystemDfContainerReport{}, map[string]string{
+		"ContainerID":  "CONTAINER ID",
+		"LocalVolumes": "LOCAL VOLUMES",
+		"RWSize":       "SIZE",
+	})
 	containerRow := "{{.ContainerID}}\t{{.Image}}\t{{.Command}}\t{{.LocalVolumes}}\t{{.RWSize}}\t{{.Created}}\t{{.Status}}\t{{.Names}}\n"
-	format = containerHeaders + "{{range . }}" + containerRow + "{{end}}"
-	if err := writeTemplate(w, format, dfContainers); err != nil {
+	format = "{{range . }}" + containerRow + "{{end}}"
+	if err := writeTemplate(cmd, w, hdrs, format, dfContainers); err != nil {
 		return nil
 	}
 
 	// Volumes
-	fmt.Print("\nLocal Volumes space usage:\n\n")
+	fmt.Fprint(w, "\nLocal Volumes space usage:\n\n")
 
 	dfVolumes := make([]*dfVolume, 0, len(reports.Volumes))
 	// convert to dfVolume for output
 	for _, d := range reports.Volumes {
 		dfVolumes = append(dfVolumes, &dfVolume{SystemDfVolumeReport: d})
 	}
-	volumeHeaders := "VOLUME NAME\tLINKS\tSIZE\n"
+	hdrs = report.Headers(entities.SystemDfVolumeReport{}, map[string]string{
+		"VolumeName": "VOLUME NAME",
+	})
 	volumeRow := "{{.VolumeName}}\t{{.Links}}\t{{.Size}}\n"
-	format = volumeHeaders + "{{range . }}" + volumeRow + "{{end}}"
-	return writeTemplate(w, format, dfVolumes)
+	format = "{{range . }}" + volumeRow + "{{end}}"
+	return writeTemplate(cmd, w, hdrs, format, dfVolumes)
 }
 
-func writeTemplate(w io.Writer, format string, output interface{}) error {
-	tmpl, err := template.New("dfout").Parse(format)
+func writeTemplate(cmd *cobra.Command, w *tabwriter.Writer, hdrs []map[string]string, format string,
```

Comment:
Maybe it's my twisted mind, and it's a nit at best, but I think any I/O function should have the I/O object as the first param, not further down the line.  i.e.: 
```suggestion
func writeTemplate(w *tabwriter.Writer, cmd *cobra.Command, hdrs []map[string]string, format string,
```
My $.02, YMMV

Reply:
I used this order because that is the order from the other cobra functions